lockfile/src/lib.rs

```diff
@@ -49,9 +49,9 @@ pub enum LockfileFormat {
     Gradle,
     // This is historically called "nuget" but it's actually for MSBuild project files.
     // Nuget has its own file formats that are not currently supported.
-    #[serde(rename = "nuget")]
-    #[serde(alias = "msbuild")]
+    #[serde(alias = "nuget")]
     Msbuild,
+    NugetLock,
```

Comment:
I'd personally much prefer to call this just `nuget` because `nugetlock` vs `nuget` is incredibly confusing. However I'm not sure we can currently make this change without breaking user's setups.

It might be best to leave this as `nugetlock` for a while until project files are transitioned to the new default `msbuild` name for `MsBuild`, so we can then claim the `nuget` identifier for this.

Reply:
There is another format where old NuGet projects use a packages.config file. Some people still use it.

Reply:
I think that's more of a manifest than lockfile, right?

Reply:
Sure enough...it appears to still be an [additional supported format](https://learn.microsoft.com/en-us/nuget/reference/packages-config). It looks mostly like a lockfile, but the `allowedVersions` field might introduce some additional complexity. That field is used for _updates_ and not _install_ or _restore_ operations.

Reply:
`packages.config` isn't a lockfile but it normally functions as much like a lockfile as you can get for `pip`. It contains the transitive dependencies and it normally contains the exact versions of dependencies that are used.

`allowedVersions` is just a limited version selector in a different format that isn't too hard to convert, but we probably don't care about it.